automation/workflow/store.go

```diff
@@ -0,0 +1,307 @@
+package workflow
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/qri-io/qri/profile"
+)
+
+// Store manages & stores workflows, allowing listing and updating of workflows
+type Store interface {
+	Lister
+	// Create creates a new Workflow and adds it to the Store. If a workflow
+	// with that dataset id already exists, it emits an
+	// `ErrWorkflowForDatasetExists` error
+	Create(did string, pid profile.ID, triggers []Trigger, hooks []Hook) (*Workflow, error)
```

Comment:
When describing interface methods, I think it's a good idea to avoid implementation-specific details like what error is returned, but focus on behavior instead.

```
// Create add a Workflow to the store. Datasets can have at most one workflow
```

Regarding the method itselfI think this should be refactored to:

```go
func Put(*Workflow) (error)
```

Breaking the fields out & constructing the structure internally requires breaking this interface whenever the Workflow struct changes. Using the workflow struct as a parameter means the store just needs to validate the structure before saving.

Reply:
Great, totally on board w/ following the usual data store pattern.